codegen/smithy-aws-typescript-codegen/src/main/java/software/amazon/smithy/aws/typescript/codegen/AddBuiltinPlugins.java

```diff
@@ -74,10 +85,30 @@
                         .withConventions(AwsDependency.ADD_GLACIER_API_VERSION.dependency,
                                          "AddGlacierApiVersion", HAS_MIDDLEWARE)
                         .servicePredicate((m, s) -> testServiceId(s, "Glacier"))
+                        .build(),
+                RuntimeClientPlugin.builder()
+                        .withConventions(AwsDependency.SSEC.dependency, "Ssec", HAS_MIDDLEWARE)
+                        .servicePredicate((m, s) -> testServiceId(s, "S3"))
+                        .operationPredicate((m, s, o) -> testContainsMember(m, o, SSEC_OPERATIONS))
                         .build()
         );
     }
 
+    private static boolean testContainsMember(Model model, OperationShape operationShape, Set<String> expectedMemberNames) {
+        ShapeIndex shapeIndex = model.getShapeIndex();
```

Comment:
We're deprecating ShapeIndex in favor of just using the Model directly. Any method you were calling on the ShapeIndex should just get called on the Model.

Reply:
fixed in 6bcea90f30616e16983326a3b87381730520931d